### PR TITLE
Fix to Block Microservices on termination signal #529

### LINKED ISSUE #529: Block Microservices on termination signal
After microservices has started block it on termination signal (such as SIGTERM for example).
Upon receiving exiting signal do microservices shutdown.

http://man7.org/linux/man-pages/man7/signal.7.html
https://docs.oracle.com/javase/10/troubleshoot/handle-signals-and-exceptions.htm#JSTGD345

### PR DESCRIPTION
**Pull request by artem-v**
Fixed #529

### CODE REVIEW

File: services/src/main/java/io/scalecube/services/Microservices.java [hunk 1]
```diff
@@ -448,7 +488,7 @@ private GatewayBootstrap addFactory(Function<GatewayOptions, Gateway> factory) {
                   .doFinally(
                       s -> {
                         if (!gateways.isEmpty()) {
-                          LOGGER.info("Gateways have been stopped");
+                          LOGGER.info("Gateways has been stopped");
```
Comment (dmytro-lazebnyi) [inline services/src/main/java/io/scalecube/services/Microservices.java]: `have`

File: services/src/main/java/io/scalecube/services/Microservices.java [hunk 2]
```diff
@@ -244,11 +258,37 @@ public ServiceDiscovery discovery() {
...
   public Mono<Void> shutdown() {
     return Mono.defer(
-        () ->
-            Mono.whenDelayError(
-                discoveryBootstrap.shutdown(),
-                gatewayBootstrap.shutdown(),
-                transportBootstrap.shutdown()));
+        () -> {
+          shutdown.onComplete();
+          return onShutdown;
```
Comment (segabriel) [inline services/src/main/java/io/scalecube/services/Microservices.java:263]: if we have the separate method, why do we need to return anything? I propose to return just `void` from this method.

Comment (artem-v) [reply services/src/main/java/io/scalecube/services/Microservices.java:263]: Because I don't want to change clients all around the code.

**Review by segabriel — APPROVED** [review]
great job!

**Review by dmytro-lazebnyi — APPROVED** [review]

**Pull request merged**